### DISCUSSION TITLE
Implementing collision testing in polygon or geoJSON layer

### DISCUSSION THREAD

**Discussion by abiswas97** [visgl/deck.gl#8068] [Q&A] [answered]
Hi! Is there currently a way to implement polygon bounds collision? I've tried using the `CollisionFilterExtension`, but this can either blur the lines of the polygons, or distort them. I'm assuming this is becuase the collision is point in polygon based?

My use case is that I have several nuclei bounding boxes annotated, and as I zoom out I would like to hide colliding bounding box polygons.

**Comment by Pessimistress** [comment, marked as answer]
`CollisionFilterExtension` is meant for point-based data. You need another library, such as [turfjs](https://turfjs.org/), to detect overlap in your use case.

**Reply by abiswas97** [reply]
Thanks! Will check this out :)
